src/pages/HomePage.jsx

```diff
@@ -2,53 +2,55 @@
 // IMPORTS
 // ============================================================================================
 
-import { useContext } from "react";
+import { useContext, useState } from "react";
 import { UserContext } from "../contexts/UserContext";
+import ResponsiveGrid from "../components/ResponsiveGrid";
+import Course from "../components/Course";
+import CourseCard from "../components/CourseCard";
 
 // ============================================================================================
 // COMPONENT DEFINITION
 // ============================================================================================
 
 function HomePage() {
   const { userInfo, setUserInfo } = useContext(UserContext);
-
-  return (
-    <>
-      <p>User information:</p>
-
-      <table border={2}>
-        <tbody>
-          <tr>
-            <th>Access Token</th>
-            <td>{userInfo.access_token}</td>
-          </tr>
-          <tr>
-            <th>Token Type</th>
-            <td>{userInfo.token_type}</td>
-          </tr>
-          <tr>
-            <th>Name</th>
-            <td>{userInfo.user_data.name}</td>
-          </tr>
-          <tr>
-            <th>E-mail</th>
-            <td>{userInfo.user_data.email}</td>
-          </tr>
-          <tr>
-            <th>Learner Data</th>
-            <td>{JSON.stringify(userInfo.user_data.learnerData)}</td>
-          </tr>
-          <tr>
-            <th>Instructor Data</th>
-            <td>{JSON.stringify(userInfo.user_data.instructorData)}</td>
-          </tr>
-        </tbody>
-      </table>
-      <br />
-
-      <button onClick={() => setUserInfo(null)}>Go back to login screen.</button>
-    </>
-  );
+  const [course, setCourse] = useState(null);
+
+  const userCourses = userInfo?.user_data?.learnerData?.map
+    ? userInfo.user_data.learnerData
+    : [];
+
+  if (course !== null) {
+    return (
+      <>
+        <Course courseData={course} />
+        <br />
+
+        <button onClick={() => setCourse(null)}>Close</button>
+      </>
+    );
+  } else {
+    return (
+      <>
+        <h1>Your Courses</h1>
+
+        <ResponsiveGrid minColumnWidth="329px" rowGap="43px">
+          {userCourses.map((course, index) => {
+            return (
+              <CourseCard
+                courseData={course}
+                onViewClick={() => setCourse(course)}
+                key={index}
+              />
+            );
+          })}
+        </ResponsiveGrid>
+        <br />
```

Comment:
Avoid excessive <br> tags for spacing; use CSS margins or padding instead.

Reply:
Agreed (this was done here as a quick-fix for the demo).